static/app/components/avatar/gravatar.tsx

```diff
@@ -15,75 +15,62 @@ type Props = {
   placeholder?: string;
 } & ImageStyleProps;
 
-type State = {
-  MD5?: any;
-};
-
-class Gravatar extends Component<Props, State> {
-  state: State = {
-    MD5: undefined,
-  };
-
-  componentDidMount() {
-    this._isMounted = true;
-
-    import('crypto-js/md5')
-      .then(mod => mod.default)
-      .then(MD5 => {
-        if (!this._isMounted) {
-          return;
-        }
-        this.setState({MD5});
-      });
-  }
-
-  componentWillUnmount() {
-    // Need to track mounted state because `React.isMounted()` is deprecated and because of
-    // dynamic imports
-    this._isMounted = false;
-  }
-
-  private _isMounted: boolean = false;
-
-  buildGravatarUrl = () => {
-    const {gravatarId, remoteSize, placeholder} = this.props;
-    let url = ConfigStore.getConfig().gravatarBaseUrl + '/avatar/';
-
-    const md5 = callIfFunction(this.state.MD5, gravatarId);
-    if (md5) {
-      url += md5;
+type HasherHelper = typeof import('crypto-js/md5');
+
+function Gravatar({
+  remoteSize,
+  gravatarId,
+  placeholder,
+  round,
+  onError,
+  onLoad,
+  suggested,
+  grayscale,
+}: Props) {
+  const isMounted = useRef(false);
+  const {gravatarBaseUrl} = useLegacyStore(ConfigStore);
+  const [MD5, setMD5] = useState<HasherHelper>();
+
+  const loadMd5Helper = useCallback(async () => {
+    const mod = await import('crypto-js/md5');
```

Comment:
Seems like this would be a bit simpler if it used Suspense instead?

Reply:
Oh yea

Reply:
this is def pre suspense, good call